tests/basic_service_test.go

```diff
@@ -159,18 +159,22 @@ func TestBasicService(t *testing.T) { // nolint
 				fromToPeer[p.Name] = endpoint
 			}
 
-			// Same client reach with different port to the session affinity service, should have same destination pod,
-			zap.L().Info(fmt.Sprintf("Testing connections to different ports of sesson affinity service, should use same from/to peers: %v", fromToPeer))
+			// to validate if affiliation applies to same port
+			zap.L().Info(fmt.Sprintf("Testing connections to same ports, try multiple times to affirm the affiliation, should use same from/to peers: %v", fromToPeer))
 			zap.L().Debug(fmt.Sprintf("Session affinity peers: %v", fromToPeer))
-			zap.L().Info("Connection via port 80")
-			reachabilityPort80 := matrix.NewReachability(pods, false)
-			for from, to := range fromToPeer {
-				reachabilityPort80.ExpectPeer(&matrix.Peer{Namespace: namespace, Pod: from}, &matrix.Peer{Namespace: namespace, Pod: to}, true)
+			for i := 0; i < 3; i++ {
```

Comment:
why 3? possible to change it as a const?

Reply:
if this part of logic is retryable / repeatable, is it possible to have some sort of generic functions or wrapper?

Reply:
3 times is enough to affirm the connection affinity. More brings marginal affirmation while extent the test lifetime.

This is not a retry here, we do not exit when succeed, we need to run thru the 3 times to affirm the affinity.

Have put "3" into a const.